### PR TITLE
B-21041-INT Prime can update second/third addresses

### PR DESCRIPTION
**Pull request by paulstonebraker**
## Summary

Part of the effort to allow prime to view/update/add third addresses (also includes the work to add second addresses). This work allows prime to update addresses for existing shipments that have second/third addresses. In planning my approach, I thought it would be quicker to refactor the existing Prime UI update address component, and make it better for those of us who use Prime SIM, so I took the liberty of doing that. Now, when you click update on an address, it will take you an update address page for that specific address.

### How to test
Setup: if [21054](https://github.com/transcom/mymove/pull/13692) has not been merged you will either need a move with shipments of all types, with second/third pickup/delivery addresses

1. Access the move described above as prime
2. On the move details screen, scroll to a shipment with addresses and click 'Edit' next to an address
3. At the new screen, confirm that the details are all accurate
4. Update the data and click save and confirm data has properly saved
5. Try with all combinations of shipments/addresses
NOTE: NTS/NTSR will not update if a primary pickup or destination address is missing, its an existing bug
NOTE: HHG Primary destination addresses will not update, as they have a separate endpoint to use
NOTE: Updating PPM second/third addresses is being handled in [21054](https://github.com/transcom/mymove/pull/13692)

### CODE REVIEW

**Review by pambecker — APPROVED** [review]

File: src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdateAddress.test.jsx [hunk 1]
```diff
@@ -179,74 +132,11 @@ describe('PrimeUIShipmentUpdateAddress page', () => {
...
-      const shipment = noPickupAddressReturnValue.moveTaskOrder.mtoShipments[shipmentIndex];
-
-      await waitFor(() => {
-        expect(shipment.pickupAddress).toBeNull();
-        expect(screen.getByRole('heading', { name: /Destination address/, level: 2 }));
-        expect(screen.getAllByLabelText('Address 1').length).toBe(1);
-        expect(screen.getAllByLabelText('Address 1')[0]).toHaveValue(shipment.destinationAddress.streetAddress1);
-        expect(screen.getAllByLabelText(/Address 2/)[0]).toHaveValue('');
-        expect(screen.getAllByLabelText('City')[0]).toHaveValue(shipment.destinationAddress.city);
-        expect(screen.getAllByLabelText('State')[0]).toHaveValue(shipment.destinationAddress.state);
```
Comment (loganwc) [inline src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdateAddress.test.jsx:248]: is your new code tested?

**Review by loganwc — CHANGES_REQUESTED** [review]

File: src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdateAddress.test.jsx (hunk 1 above)
Comment (paulstonebraker) [reply src/pages/PrimeUI/Shipment/PrimeUIShipmentUpdateAddress.test.jsx:248]: Yes. I refactored this component to display/allow to edit the address you click "edit" on. The previous version had both pickup and destination addresses on it, with separate save buttons that would save that specific address, and return to move details screen.

File: pkg/handlers/primeapi/mto_shipment_address.go [hunk 2]
```diff
@@ -39,7 +39,7 @@ func (h UpdateMTOShipmentAddressHandler) Handle(params mtoshipmentops.UpdateMTOS
 			}
 
 			if dbShipment.ShipmentType == models.MTOShipmentTypeHHGIntoNTSDom &&
-				(*dbShipment.DestinationAddressID == addressID) {
+				(dbShipment.DestinationAddressID == nil || *dbShipment.DestinationAddressID == addressID) {
```
Comment (paulstonebraker) [inline pkg/handlers/primeapi/mto_shipment_address.go:42]: @ajlusk I updated this and the NTSR below to allow for clearer error messaging when a move does not yet have an existing destination (for NTS) or pickup (for NTSR) address... do you see this check causing any issues that I'm not thinking of?

Comment (ajlusk) [reply pkg/handlers/primeapi/mto_shipment_address.go:42]: No concerns come to mind. Covers a scenario I missed. ✅

File: pkg/handlers/primeapi/mto_shipment_address.go [hunk 3]
```diff
@@ -39,19 +39,19 @@ func (h UpdateMTOShipmentAddressHandler) Handle(params mtoshipmentops.UpdateMTOS
 			}
 
 			if dbShipment.ShipmentType == models.MTOShipmentTypeHHGIntoNTSDom &&
-				(dbShipment.DestinationAddressID == nil || *dbShipment.DestinationAddressID == addressID) {
+				(dbShipment.DestinationAddressID != nil || *dbShipment.DestinationAddressID == addressID) {
```
Comment (pambecker) [inline pkg/handlers/primeapi/mto_shipment_address.go:42]: I believe we want these to be && if statements don't we?

**Review by loganwc — APPROVED** [review]

**Comment by pambecker** [comment]
All code looks good and is working according to steps listed.  Just need the AC in the Agility story updated, believe this is waiting on word with Terri.

**Review by pambecker — APPROVED** [review]

**Pull request merged**